src/validation.cpp

```diff
@@ -2738,7 +2738,12 @@ void CChainState::PruneBlockIndexCandidates()
 
     auto temp_set = setBlockIndexCandidates;
     for (const auto& el : temp_set) {
-        if (el->pprev != nullptr) {
+        if(m_chain.Contains(el) && el != m_chain.Tip()) {
+            setBlockIndexCandidates.erase(el);
+            continue;
+        }
```

Comment:
I'd suggest to make the arguments order same in both ifs:
`
if(m_chain.Tip() != el && m_chain.Contains(el)) {
`